src/main.ts

```diff
@@ -1,8 +1,16 @@
+import { RequestMethod, VersioningType } from "@nestjs/common";
 import { NestFactory } from "@nestjs/core";
 import { AppModule } from "./app.module";
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
+  app.setGlobalPrefix("api/v1", {
```

Comment:
This is not a valid approach for versioning in nestjs.
This will add "api/v1" prefix for all route. By following nest js convention of versioning at controller level your path will look like this:
`api/v1/v1/user`
`api/v1/v2/user`

This is not the required result for us.